registry/cmd/main.go

```
@@ -0,0 +1,98 @@
+		},
+		LeaderElection:   true,
+		LeaderElectionID: "registry-provisioner.opencloud.wso2.com",
+	})
+	if err != nil {
+		logger.Fatal("failed to create controller manager", zap.Error(err))
+	}
+
+	if err := (&controller.RegistryBackendReconciler{
+		Client:   mgr.GetClient(),
+		Scheme:   mgr.GetScheme(),
+		Recorder: mgr.GetEventRecorderFor("registrybackend"),
```

golangci-lint (2.12.2)

[error] SA1019: mgr.GetEventRecorderFor is deprecated: this uses the old events API and will be removed in a future release. Please use GetEventRecorder instead. 

(staticcheck)